src/main/kotlin/uk/gov/justice/hmpps/offendersearch/security/AuthAwareTokenConverter.kt

```diff
@@ -13,18 +13,13 @@ class AuthAwareTokenConverter : Converter<Jwt, AbstractAuthenticationToken> {
   private val jwtGrantedAuthoritiesConverter: Converter<Jwt, Collection<GrantedAuthority>> = JwtGrantedAuthoritiesConverter()
 
   override fun convert(jwt: Jwt): AbstractAuthenticationToken {
-    val claims = jwt.claims
-    val principal = findPrincipal(claims)
-    val authorities = extractAuthorities(jwt)
-    return AuthAwareAuthenticationToken(jwt, principal, authorities)
-  }
-
-  private fun findPrincipal(claims: Map<String, Any?>): String {
-    return if (claims.containsKey("user_name")) {
-      claims["user_name"] as String
-    } else {
-      claims["client_id"] as String
-    }
+    return AuthAwareAuthenticationToken(
+        jwt = jwt,
+        clientOnly = jwt.subject == jwt.claims["client_id"],
+        subject = jwt.subject,
+        deliusUser = "delius" == jwt.claims["auth_source"],
```

Comment:
important, since the exclusion/inclusion list will only work for Delius users (it uses Delius USER_ table foreign key)